server/container/core/src/main/java/org/apache/james/server/core/MailImpl.java

```diff
@@ -256,18 +264,15 @@ public Builder addAllHeadersForRecipients(PerRecipientHeaders perRecipientHeader
         }
 
         public MailImpl build() {
-            MailImpl mail = new MailImpl();
+            MailImpl mail = new MailImpl(state.orElse(DEFAULT), attributes, recipients, perRecipientHeaders);
```

Comment:
are you using the state as a mail name? Maybe it's type to add some strong typing on state and name.

Reply:
Not really doable

Reply:
why?

Reply:
MailImpl implements Mail interface that is part of the public mailet API.
The spread of the strongly typed Name and State would be so short that it doesn't make sense in my opinion.
But of course I may be wrong in my analysis